Update Getting Started VRF tutorial from v1 to v2 (#689)
smartcontractkit/documentation: docs/Introduction/tutorials/intermediates-tutorial.md

  <a href="/docs/conceptual-overview/#what-is-remix" >What is Remix?</a>
</div>

You have now completed all necessary functions to generate randomness and assign the user a _Game of Thrones_ house. We've added a few helper functions in there to make using the contract easier and more flexible. You can deploy and interact with the complete contract in Remix.

## 5. How do I deploy to testnet?

You will now deploy your completed contract. This deployment is slightly different than the example in the [Deploy Your First Contract](/docs/deploy-your-first-contract/) guide. In our case, you will have to pass in parameters to the constructor upon deployment.

Once compiled, you'll see a dropdown menu that looks like this in the deploy pane:

![Remix contract selected](/files/intermediates-tutorial-01.png)

Select the `VRFD20` contract or the name that you gave to your contract. You will deploy this contract on the Rinkeby test network.

Click the caret arrow on the right hand side of **Deploy** to expand the parameter fields, and paste your subscription id.

> ![Remix contract parameters to deploy](/files/intermediates-tutorial-02.png)

Reviewer: You might want to scale some of the larger images down a little bit if you have time.

Then click the `Deploy` button and use your Metamask account to confirm the transaction.

**Note**: You should [have some Rinkeby ETH](/docs/deploy-your-first-contract/#install-and-fund-your-metamask-wallet) in your Metamask account to pay for the GAS.

> 📘 Address, Key Hashes and more
>
> For a full reference of the addresses, key hashes and fees for each network, see [VRF Contract Addresses](/docs/vrf-contracts/#configurations).

At this point, your contract should be successfully deployed. However, it can't request anything yet since it is not whitelisted in the [Subscription Manager](https://vrf.chain.link). If you click `rollDice`, the transaction will revert.

## 6. How do I add my contract to my subscription account?

Once you deploy your contract, it should be whitelisted in the subscription account so it can use the funding when requesting for randomness. Go to the [Subscription Manager](https://vrf.chain.link) and add your deployed contract address to the list of consumers. Find your contract address in Remix under **Deployed Contracts** on the bottom left.

![Remix contract address](/files/intermediates-tutorial-03.png)

## 7. How do I test `rollDice`?

> After you open the deployed contract tab in the bottom left, the function buttons are available. Find `rollDice` and click the caret to expand the parameter fields. Enter an Ethereum address, and click 'rollDice'.

Reviewer: ```suggestion
After you open the deployed contract tab in the bottom left, the function buttons are available. Find `rollDice` and click the caret to expand the parameter fields. Enter an Ethereum address to specify a "dice roller", and click 'rollDice'.
```

> You will have to wait a few minutes for your transaction to confirm and the response to be sent back. You can get your house by clicking the `house` function button with the address passed in `rollDice`. Once the response has been sent back, you'll be assigned a _Game of Thrones_ house!

Reviewer: ```suggestion
You will have to wait a few minutes for your transaction to confirm and the response to be sent back. You can get your house by clicking the `house` function button with the address passed in `rollDice`. After the response is sent back, you'll be assigned a _Game of Thrones_ house!
```
